cli/src/main/java/com/devonfw/tools/ide/tool/aws/Aws.java

```diff
@@ -0,0 +1,72 @@
+package com.devonfw.tools.ide.tool.aws;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Set;
+
+import com.devonfw.tools.ide.cli.CliException;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.environment.EnvironmentVariables;
+import com.devonfw.tools.ide.environment.EnvironmentVariablesType;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.ToolCommandlet;
+
+import static com.devonfw.tools.ide.os.OperatingSystem.LINUX;
+
+/**
+ * {@link ToolCommandlet} for AWS CLI (aws).
+ */
+
+public class Aws extends ToolCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Aws(IdeContext context) {
+
+    super(context, "aws", Set.of(TAG_CLOUD));
+  }
+
+  @Override
+  public void postInstall() {
+
+    super.postInstall();
+
+    if (this.context.getSystemInfo().getOs() == LINUX) {
+      ProcessContext pc = this.context.newProcess();
+      Path installToolPath = this.getToolPath();
+      Path linuxInstallScript = installToolPath.resolve("install");
+      pc.executable(linuxInstallScript);
+      pc.addArgs("-i", installToolPath.toString(), "-b", installToolPath.toString());
+      pc.run();
+
+      // I (MattesMrzik) used WSL to test this and my aws binary was not marked executable by default.
+      pc.executable("chmod");
+      Path realPathOfBinary = null;
+      try {
+        realPathOfBinary = installToolPath.resolve("aws").toRealPath();
+      } catch (IOException e) {
+        throw new CliException(
+            "Failed to get real path of " + installToolPath.resolve("aws") + " for making it executable.");
+      }
+      pc.addArgs("+x", realPathOfBinary.toString());
+      pc.run();
+
+      this.context.getFileAccess().delete(linuxInstallScript);
+      this.context.getFileAccess().delete(installToolPath.resolve("dist"));
+    }
```

Comment:
Great that you are testing for and considering all platforms.
However, we distinguish local and global tools. For global tools, we just download and run an installer.
However, for local tools already the installation logic shall create the proper installation.
It is still great to reuse our existing code and logic to download and extract tools but IMHO this is the proper place and way to solve it:
Here on linux the actual generic installation process with complete but only have an "installed" in `software/aws` and also create a `.ide.tool.version`. Therefore if the post installation fails, it will never recover from its inconsistent state.
Even worse the installer has already installed in the software repository and your post installation logic is somehow "in-situ" manipulating this result. Therefore when installing version X then installing Y and then installing version X again will result in symlinking the previous X version from the software repository where post install has already run on. So IMHO this will lead to a dead end and will never get stable and free of bugs.

Instead if we can not get a proper installation package for linux, we have to hook in way earlier.
Instead of moving the extracted contents to `$IDE_ROOT/software/default/aws/aws/«version»` we have to there run the installer to do the installation and write it there.

I am unsure if `aws` is the only commandlet where we have such problem.
If there are multiple such commandlets, we should create some more generic and reusable approach in the super-class hierarchy to make this happen easily in the commandlets. This somehow conflicts with #47 that is still WIP and not yet completed and merged. So feel invited to exchange with @salimbouch about this and PR #113.

Reply:
Thanks for the remark. I am thinking about adding an extra "if" clause to the `ToolCommandlet::extract() `method. For aws the `./install` script takes arguments `-i` as install dir path and `-b` the directory to store symlinks to executables. This might not be consistent across tools that are installed with their own script. So maybe instead of adding an extra "if" to `ToolCommandlet::extract()` Aws.java should overwrite this method.

Reply:
I now added an extra method `moveAndProcessExtraction `to ToolCommandlet that moves the extraction content from the `tmp` dir to the `software` dir. Usually this is just a call to FileAccess.move, but in the case for `aws` where an install script has to be called (on linux), `moveAndProcessExtraction` also calls this script.